mmdet/models/backbones/hrnet.py

```diff
@@ -0,0 +1,498 @@
+        conv_cfg (dict): dictionary to construct and config conv layer.
+        norm_cfg (dict): dictionary to construct and config norm layer.
+        norm_eval (bool): Whether to set norm layers to eval mode, namely,
+            freeze running stats (mean and var). Note: Effect on Batch Norm
+            and its variants only.
+        with_cp (bool): Use checkpoint or not. Using checkpoint will save some
+            memory while slowing down the training speed.
+        zero_init_residual (bool): whether to use zero init for last norm layer
+            in resblocks to let them behave as identity.
+    """
+
+    blocks_dict = {
+        'BASIC': BasicBlock,
+        'BOTTLENECK': Bottleneck
+    }
+
+    def __init__(self,
+                 extra,
+                 conv_cfg=None,
+                 norm_cfg=dict(type='BN'),
+                 norm_eval=True,
+                 with_cp=False,
+                 zero_init_residual=False,):
+        super(HRNet, self).__init__()
+        self.norm_eval = norm_eval
+        self.norm_cfg = norm_cfg
+        self.conv_cfg = conv_cfg
+        self.with_cp = with_cp
+        self.zero_init_residual = zero_init_residual
+        # for
+        self.extra = extra
+        # stem network
+        # stem net
+        self.norm1_name, norm1 = build_norm_layer(self.norm_cfg, 64,
+                                                  postfix=1)
+        self.norm2_name, norm2 = build_norm_layer(self.norm_cfg, 64,
+                                                  postfix=2)
+
+        self.conv1 = build_conv_layer(self.conv_cfg, 3, 64,
+                                      kernel_size=3,
+                                      stride=2,
+                                      padding=1,
+                                      bias=False)
+
+        self.add_module(self.norm1_name, norm1)
+        self.conv2 = build_conv_layer(self.conv_cfg, 64, 64,
+                                      kernel_size=3,
+                                      stride=2,
+                                      padding=1,
+                                      bias=False)
+
+        self.add_module(self.norm2_name, norm2)
+        self.relu = nn.ReLU(inplace=True)
+
+        # stage 1
+        self.stage1_cfg = self.extra['stage1']
+        num_channels = self.stage1_cfg['num_channels'][0]
+        block_type = self.stage1_cfg['block']
+        num_blocks = self.stage1_cfg['num_blocks'][0]
+
+        block = self.blocks_dict[block_type]
+        stage1_out_channels = num_channels * block.expansion
+        self.layer1 = self._make_layer(block, 64, num_channels, num_blocks)
+
+        # stage 2
+        self.stage2_cfg = self.extra['stage2']
+        num_channels = self.stage2_cfg['num_channels']
+        block_type = self.stage2_cfg['block']
+
+        block = self.blocks_dict[block_type]
+        num_channels = [num_channels[i] * block.expansion
+                        for i in range(len(num_channels))]
```

Comment:
`i` is unused, so just `num_channels = [channel * block.expansion for channel in num_channels]`